### PR TITLE
opensmtpd: 6.4.2p1 -> 6.6.1p1

### PR DESCRIPTION
**Pull request by flokli**
The substitition in smtpd/parse.y isn't necessary anymore (it's gone).

opensmtpd 6.6.0p1 fails to build with libressl like this:
```
    make[1]: Entering directory '/build/opensmtpd-6.6.0p1'
    Making all in openbsd-compat
    make[2]: Entering directory '/build/opensmtpd-6.6.0p1/openbsd-compat'
    gcc -DHAVE_CONFIG_H -I. -I..  -I../smtpd -I../openbsd-compat -I../openbsd-compat/err_h -I/nix/store/7hgyarrzkhvp47ynwljabdfy28vzwjwb-libevent-2.1.11-dev/include   -ffunction-sections -c -o arc4random.o arc4random.c
    arc4random.c:167:21: error: macro "arc4random_stir" passed 1 arguments, but takes just 0
     arc4random_stir(void)
                         ^
    arc4random.c:168:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
     {
     ^
    make[2]: *** [Makefile:445: arc4random.o] Error 1
    make[2]: Leaving directory '/build/opensmtpd-6.6.0p1/openbsd-compat'
    make[1]: *** [Makefile:419: all-recursive] Error 1
    make[1]: Leaving directory '/build/opensmtpd-6.6.0p1'
    make: *** [Makefile:351: all] Error 2
```
Internet suggests it's a libressl-specific issue, so build with openssl for now.

Also update the module to use tmpfiles to set up the spool directories.
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @Ekleog @rickynils @obadz

### CODE REVIEW

**Review by Ekleog — DISMISSED** [review]
One important comment about the removed substitution below.

Also, I seem to remember from the mailing list that 6.6.0p2 is underway/almost coming, fixing issues with some platforms -- don't know whether it'd be worth it to wait for p2 to do the upgrade here, though.

@ofborg build opensmtpd.tests

File: pkgs/servers/mail/opensmtpd/default.nix [hunk 1]
```diff
@@ -21,8 +21,6 @@ stdenv.mkDerivation rec {
   # See https://github.com/OpenSMTPD/OpenSMTPD/issues/885 for the `sh bootstrap`
   # requirement
   postPatch = ''
-    substituteInPlace smtpd/parse.y \
-      --replace "/usr/libexec/" "$out/libexec/opensmtpd/"
```
[the remark this answers is not in the document]
Comment (flokli) [reply pkgs/servers/mail/opensmtpd/default.nix]: nice catch! Indeed we should patch that too. Any idea why the nixos test didn't uncover that?

[the remark this answers is not in the document]
Comment (flokli) [reply pkgs/servers/mail/opensmtpd/default.nix]: Ah, the `Makefile`'s set `-DPATH_LIBEXEC=\"$(pkglibexecdir)\"`, which gets set by the build system properly - so this substitution can be skipped.

**Comment by flokli** [comment]
Hm, we probably should hold this off until 6.6.0p2, and check building with libressl then: https://github.com/OpenSMTPD/OpenSMTPD/issues/958.

File: pkgs/servers/mail/opensmtpd/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ekleog) [reply pkgs/servers/mail/opensmtpd/default.nix]: Oh cool, hadn't noticed :)

**Comment by flokli** [comment]
I pushed a new version pointing to the latest version of the "portable" branch, which seems to be able to build work with libressl again.

**Comment by Ekleog** [comment]
Must say I don't really feel great about releasing as 6.6.0p1 something that's actually halfway between p1 and p2 -- IMO it'd be better to either release p1 with openssl or wait for p2 with libressl?

**Comment by flokli** [comment]
No, I didn't meant to merge this as it is now, but couldn't find a button to degrade this to a draft PR - we should wait for the 6.6.0p2 release.

**Comment by Ekleog** [comment]
Sounds great to me, thanks!

**Comment by Ekleog** [comment]
`6.6.1p1` appears to be out, which I guess is what I was expecting to be released as `6.6.0p2` :)

**Comment by flokli** [comment]
Bumped to the latest official release, and refactored the test to the python test driver. (cc https://github.com/NixOS/nixpkgs/issues/72828) PTAL.

**Comment by Ekleog** [comment]
@ofborg build opensmtpd.test

**Comment by Ekleog** [comment]
@ofborg build opensmtpd.tests

**Review by Ekleog — APPROVED** [review]
Looks great to me! Letting you merge when you feel best :)

**Pull request merged**